internal/zinc-core/src/main/scala/sbt/internal/inc/UsedName.scala

```diff
@@ -14,9 +14,15 @@ import xsbti.UseScope
 case class UsedName(name: String, scopes: util.EnumSet[UseScope])
 
 object UsedName {
+
+  private def escapeControlChars(name: String) = {
+    name.replaceAllLiterally("\n", "\u26680A")
+  }
+
   def apply(name: String, scopes: Iterable[UseScope] = Nil): UsedName = {
+    val escapedName = escapeControlChars(name)
```

Comment:
This is only done for the `apply` in `UsedName`. If there's some part of the code that uses the constructor of the case class directly, the name will not be escaped. Can you make sure that the constructor is not used and make it private?